### DISCUSSION TITLE
A continuous laser with uniform intensity distribution

### DISCUSSION THREAD

**Discussion by Hatisa** [epochpic/epoch#727] [Q&A] [answered]
I need to simulate a continuous laser with a spot radius of 2.5 micrometers, and its spatial intensity distribution is uniform. How should I set it up?
![20241021143530](https://github.com/user-attachments/assets/4bea09a7-5947-4cf9-b6a8-efe29eb45a65)
![20241021143548](https://github.com/user-attachments/assets/0213c7e3-4ad3-4f36-8b71-cd212285d164)
I have tried to change the x_spot but it doesn't work.
laser intensity a =  (e * sqrt((Ey+Bz).^2 + (Ez-By).^2))/(2 * m_e * c * omega_L)
The results I run are as follows:
![total_result0010 sdf](https://github.com/user-attachments/assets/a959960b-0502-413e-98cc-11b1b332e6da)
I need a continuous laser like this:
![1](https://github.com/user-attachments/assets/f2d3ae76-03e2-4d97-b33e-974982f85ee0)
Thanks!

**Comment by Status-Mirror** [comment, marked as answer]
Hi @Hatisa 

Your laser wavelength is 1 micron, and your profile is 2 microns - such a laser pulse will not physically propagate the way you want it to.

Lasers move through space by following Maxwell's equations. If you set up a profile like you describe, where the beam size is comparable to the wavelength, then the kind of diffraction you observe is guaranteed - this is what Maxwell's equations produce.

To enforce a fixed intensity over some distance, you might want to consider using a 1D simulation - these are diffraction-free, as there is no variation in the $y$ direction.

If it's important that the beam has a set size at a particular point in space (e.g. a laser with a 2 micron spot-size when it hits the target), then you can use a [focusing Gaussian beam](https://epochpic.github.io/quickstart/basic_examples/laser.html#focussing_a_gaussian_beam).

Hope this helps,
Stuart
